### PR TITLE
Updating to Realm Java 3.1.2. Removing io.realm.internal test.

### PR DESCRIPTION
**Pull request by kneth**

### CODE REVIEW

**Comment by kneth** [comment]
@cmelchior Good idea but then I need to do a more controlled run of the benchmark.

**Comment by kneth** [comment]
@nhachicha We never publish it, I don't think we should promote using any `io.realm.internal` classes. Besides that, we would need to do some work to make that test work again.

File: app/build.gradle [hunk 1]
```diff
@@ -4,7 +4,7 @@ apply plugin: 'realm-android'
 
 android {
     compileSdkVersion 24
-    buildToolsVersion "24.0.0"
+    buildToolsVersion '25.0.0'
```
Comment (bmeike) [inline app/build.gradle]: I think this should be, at least, 25.0.2.  Probably 25.0.3

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Wed Apr 10 15:27:10 PDT 2013
+#Wed Apr 19 09:59:48 CEST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.13-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
```
Comment (bmeike) [inline gradle/wrapper/gradle-wrapper.properties]: I think gradle should be at 3.4.1

**Pull request merged**